### ISSUE TITLE
how M is mapped to N

### ISSUE THREAD

**Issue by dorost1234** [pengyuLPY/Virtual-Fully-Connected-Layer#3]
Hi
From the equation 1, you have y ∈ R^M, if original linear layer is of size DxN, then I am not sure how M is mapped to M, could you please let me know how this mapping is done? thanks

**Comment by pengyuLPY** [comment]
Hi:
I am glad to introduce the process to you in details. There are two parts in our code are used to mapping N to M:
    1. Line54-Line108 in Virtual-Fully-Connected-Layer/CNNs/DictarcFace.py: It generates a new 'gt-label' which is denoted as label_new. The label_new ∈ R^M. The label_new is returned to Line43 in Virtual-Fully-Connected-Layer/train_test.py and used to calculate the loss function in Line46. In this part, the label  ∈ R^N is mapped to the label_new ∈ R^M.
2.  Line14-Line15 in Virtual-Fully-Connected-Layer/config.py. We make m = n%M simply (0<=m<=M, 0<=n<=N) to group the N identities to M groups in the training dataset.

**Comment by dorost1234** [comment]
Hi
thanks for the response, so I understood you first map N to M, then train with M, but then in the test time, how would you map back  M to N? In this case, your paper can only be applied to softmax and not the linear layer, am I right on this? thanks
